homeassistant/components/axis/binary_sensor.py

```diff
@@ -62,24 +63,46 @@ def async_create_entity(event: Event) -> None:
     )
 
 
-class AxisBinarySensor(AxisEventBase, BinarySensorEntity):
+class AxisBinarySensor(AxisEventEntity, BinarySensorEntity):
     """Representation of a binary Axis event."""
 
     def __init__(self, event: Event, device: AxisNetworkDevice) -> None:
         """Initialize the Axis binary sensor."""
         super().__init__(event, device)
-        self.cancel_scheduled_update = None
+        self.cancel_scheduled_update: Callable[[], None] | None = None
 
-        self._attr_device_class = DEVICE_CLASS.get(self.event.group)
+        self._attr_device_class = DEVICE_CLASS.get(event.group)
         self._attr_is_on = event.is_tripped
 
-    @callback
-    def update_callback(self, no_delay=False):
-        """Update the sensor's state, if needed.
+        if (
+            event.group == EventGroup.INPUT
+            and event.id in self.device.api.vapix.ports
+            and self.device.api.vapix.ports[event.id].name
+        ):
+            self._attr_name = self.device.api.vapix.ports[event.id].name
+
+        if event.group == EventGroup.MOTION:
+
+            for event_topic, event_data in (
+                (EventTopic.FENCE_GUARD, self.device.api.vapix.fence_guard),
+                (EventTopic.LOITERING_GUARD, self.device.api.vapix.loitering_guard),
+                (EventTopic.MOTION_GUARD, self.device.api.vapix.motion_guard),
+                (EventTopic.OBJECT_ANALYTICS, self.device.api.vapix.object_analytics),
+                (EventTopic.MOTION_DETECTION_4, self.device.api.vapix.vmd4),
+            ):
 
-        Parameter no_delay is True when device_event_reachable is sent.
-        """
-        self._attr_is_on = self.event.is_tripped
+                if (
+                    event.topic_base == event_topic
+                    and event_data
+                    and event.id in event_data
+                ):
+                    self._attr_name = f"{self._event_type} {event_data[event.id].name}"
+                    break
```

Comment:
Instead of removing the `name` property function and inlining it here, consider renaming it to `_set_name` or something like that and calling it.

Reply:
Fixed